### DISCUSSION TITLE
Concentration dependent anisotropic elastic modulus

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#19932] [Q&A Modules: General] [answered]
Hello MOOSE experts

I am solving a coupled chemo-mechanical problem for an anisotropic material. In my material system, the elastic modulus tensor needs to be concentration-dependent. Can you please suggest the procedure to do it? Kindly suggest some useful examples as well?

I want to make linear variation of elastic tensor with concentration.
**C**(c) = (1-c)**C0**
```
[Materials]
  [./elasticity_tensor_with_Euler]
    type = ComputeElasticityTensorCP
    fill_method = symmetric9
    C_ijkl = '1.984e9 0.196e9 0.176e9 1.684e9 0.176e9 1.584e9 0.754e9 0.754e9 0.754e9'
    read_prop_user_object = euler_angle_read
  [../]
  [thermalexpansion]
    type = ComputeThermalExpansionEigenstrain
    temperature = diffused
    thermal_expansion_coeff = 3.496e-6
    stress_free_temperature = 0.0
    eigenstrain_name = thermal_strain
  []
  [stress]
    type = ComputeFiniteStrainElasticStress
  []
[]
```

Please help.

Thank you very much.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Seems like this is what you want to use:

https://mooseframework.inl.gov/source/materials/ComputeConcentrationDependentElasticityTensor.html

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello Guillaume

Thank you very much. This is exactly what I was looking for.

Just one query. Can I read the Euler angles from the data file, the same way I was doing earlier to rotate the Elastic tensor?

Thanks

**Reply by GiudGiud** [reply]
No, looks like this one only does constant euler angles. 
You could create a new class and blend the two functionalities?

**Reply by avtarsinghh1991** [reply]
Hello Guillaume

Thank you very much. I will merge the two functionalities.

Best
Avtar
